astrogrism/grism_observation.py

```diff
@@ -61,6 +67,13 @@ def __init__(self, grism_image, direct_image=None, telescope=None, instrument=No
                 self.filter = self.grism_header["FILTER"]
             elif "FILTER1" in self.grism_header:
                 self.filter = self.grism_header["FILTER1"]
+            # Check to make sure we didn't get an Fxxx instead of Gxxx
+            if self.filter[0] == "F":
+                # try aperture
+                if "APERTURE" in self.grism_header:
+                    if self.grism_header["APERTURE"][0] == "G":
+                        self.filter = self.grism_header["APERTURE"]
+                        self.filter = self.filter.replace("-REF", "")
```

Comment:
this feels hacky, but I'm not sure I can think of a better suggestion.  What is the intended consequence if either of the two nested if-statements are not true?  Just leave `self.filter` as is or do we need to raise an error?

Reply:
I agree that it does feel hacky, but I haven't determined any rhyme or reason as to whether the grism will be specified in `FILTER` or `APERTURE`, I had thought it was always in `FILTER` or `FILTER1` until I encountered this case. I think I'll check explicitly for the good case of `G`-something instead of one of perhaps many possible bad cases.